tests/streamlit_ui_tests/graph.spec.ts

```diff
@@ -514,12 +485,97 @@ test.describe('Graph Functionality', () => {
 
   test('Change font color of the graph title', async ({ page }) => {
     const mito = await getMitoFrameWithTypeCSV(page);
-    await clickButtonAndAwaitResponse(page, mito, { name: 'Graph' })
+    await openGraphEditor(mito, page);
 
     await clickTab(page, mito, 'Format');
     await mito.getByText('Text Fill').click();
     await mito.locator('#color-picker-').fill('#1c5bd9');
     await mito.locator('.mito-toolbar-bottom').click();
     await expect(mito.getByText('Column1 bar chart')).toHaveCSS('fill', 'rgb(42, 63, 95)');
   });
+
+  test('Update x axis columns', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await openGraphEditor(mito, page);
+
+    await addColumnToAxis(mito, page, 'X', 'Column2');
+    await expect(mito.locator('.g-xtitle', { hasText: 'Column2' })).toBeVisible();
+
+    await addColumnToAxis(mito, page, 'X', 'Column3');
+    await expect(mito.locator('.g-gtitle', { hasText: 'Column2,  Column3, Column1 bar chart' })).toBeVisible();
+  });
+
+  test('Update y axis columns', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await openGraphEditor(mito, page);
+
+    await addColumnToAxis(mito, page, 'Y', 'Column2');
+    await expect(mito.locator('.g-gtitle', { hasText: 'Column1,  Column2 bar chart' })).toBeVisible();
+
+    await addColumnToAxis(mito, page, 'Y', 'Column3');
+    await expect(mito.locator('.g-gtitle', { hasText: 'Column1,  Column2,  Column3 bar chart' })).toBeVisible();
+  });
+
+  test('deleting the source data deletes the graph', async ({ page }) => {
+    const mito = await getMitoFrameWithTestCSV(page);
+    await openGraphEditor(mito, page);
+
+    await mito.locator('.footer').getByText('test' ).click();
+    await mito.locator('.tab', { hasText: 'test'}).click({ button: 'right' });
+    await mito.locator('.mito-dropdown-item', { hasText: 'Delete' }).click();
+    await awaitResponse(page);
+    await expect(mito.getByText('Delete Sheet and Dependant Graphs')).toBeVisible();
+    await mito.getByText('Delete Sheet and Graphs').click();
+    await awaitResponse(page);
+    await expect(mito.getByRole('button', { name: 'Import Files' })).toBeVisible();
```

Comment:
Maybe we should check that there is not a graph tab in the footer? Here, we are only really checking that we have an empty tab